lbclient.go

```diff
@@ -91,9 +94,42 @@ func (cc *LBClient) init() {
 	}
 }
 
+// AddClient adds a new client to the balanced clients
+// returns the new total number of clients
+func (cc *LBClient) AddClient(c BalancingClient) int {
+	cc.mu.Lock()
+	cc.cs = append(cc.cs, &lbClient{
+		c:           c,
+		healthCheck: cc.HealthCheck,
+	})
+	cc.mu.Unlock()
+	return len(cc.cs)
+}
+
+// RemoveClients removes clients using the provided callback
+// if rc returns true, the passed client will be removed
+// returns the new total number of clients
+func (cc *LBClient) RemoveClients(rc func(BalancingClient) bool) int {
+	cc.mu.Lock()
+	n := 0
+	for _, cs := range cc.cs {
+		if rc(cs.c) {
+			cs = nil
+			continue
+		}
+		cc.cs[n] = cs
+		n++
+	}
```

Comment:
```suggestion
			continue
		}
		cc.cs[n] = cs
		n++
	}
	for i := n; i < len(cc.cs); i++ {
		cc.cs[i] = nil
	}
```

Reply:
I mean like this so that the `cc.cs` entries that you remove from the slice really aren't referenced anymore.